lua/entities/gmod_wire_expression2/core/custom/tracesystem.lua

```diff
@@ -216,47 +216,21 @@ local function RayCircleIntersection( Start, Dir, Pos, Normal, Radius )
 end
 
 local function RaySphereIntersection( Start, Dir, Pos, Radius )
-
-	-- New ray-sphere intersection code
-	--[
-	local A = 2 * Length(Dir)^2
-	local B = 2 * Dot(Dir,Start - Pos)
-	local C = Length(Pos)^2 + Length(Start)^2 - 2 * Dot(Pos,Start) - Radius^2
-
-	local BAC4 = B^2-(2*A*C)
-
-	if (BAC4 >= 0 and B < 0) then
-		-- Enter sphere
-		return Start + ((-sqrt(BAC4) - B) / A)*Dir
-
-		-- Exits sphere
-		-- return Start + ((sqrt(BAC4) - B) / A)*Dir
-	end
-	--]]
-
-	-- Old ray-sphere intersection code (slower)
-	--[[
-	local Normal = Dir * -1
-	local B = Normal:Dot(Pos-Start) / Normal:Dot(Dir)
-
-	-- It is a ray, not line, for a line segment, just add a distance chek.
-	if (B >= 0) then
-
-		local HitPos = Start + Dir * B
-
-		local Dist = Distance(Pos, HitPos)
-
-		if (Dist < Radius) then
-			HitPos = HitPos - Dir * sqrt(Radius ^ 2 - Dist ^ 2)
-
-			return HitPos
-		end
-
-	end
-	--]]
-
-	return false
-
+	local A = Dir:LengthSqr()

```

Comment:
Need to check if A is 0 actually since line 226 will divide by 0 otherwise.

Reply:
Fixed in: https://github.com/wiremod/wire-extras/pull/122/commits/775c7897a3bf7b0edc4807287c961ee7563bd4b7